tox.ini

```
@@ -61,7 +61,7 @@ description = Run integration tests
 deps =
     juju
```

@jaimesouza after some tinkering with how to attach NHC to slurmd, and playing around with test dependency versions, pinning python-libjuju to version 3.1.2.0 unblocks our CI pipeline. The newest version of python-libjuju, 3.3.x, changes the implementation for how resources are uploaded and it currently doesn't work with binary files. I have opened a complimenting issue here: https://github.com/juju/python-libjuju/issues/1000

For right now, we can get away with using this version of the Python API, so let's pin to it until the bug report is addressed!

```suggestion
    juju==3.1.2.0
```